### ISSUE TITLE
Iptables not persistent

### ISSUE THREAD

**Issue by janpascal** [denyhosts/denyhosts#46]
Upon a server restart, the hosts previously blocked by iptables are unblocked. I think there should be some code that recreates the iptables rules upon startup, but that may create them double when just the daemon is restarted (and not the server, too).

**Comment by slicer69** [comment]
The duplication could be an issue. I'm not sure if iptabels is smart enough to avoid duplicate rule entries. With PF it's pretty easy because PF just keeps a list of IP addresses to be blocked and throws away duplicate entries. I don't think iptables will do the same thing for us.

**Comment by jasontibbitts** [comment]
I believe the "proper" way to do this is to have denyhosts add the rules on a separate chain, and clear and regenerate that chain at each restart.

Plus isn't there a ticket for using ipset instead?  That works pretty much like you say PF works.

Yes, that's #18.  And this one is a duplicate of #37.

**Comment by slicer69** [comment]
That makes sense. Anyone want to tackle this? I do most of my development and testing of FreeBSD, which doesn't use iptables.

**Comment by jasontibbitts** [comment]
Well, the ipset stuff is already committed, isn't it?  I guess it went onto a branch and wasn't pulled into master but it might have been.  I can try to find the time to test it out (at the same time I'm getting the fedora packages all updated and trying to figure out why I can't stop the journal reading code from just randomly hanging).

**Comment by slicer69** [comment]
You're right, the ipset code is in its own branch. It was committed a while back, but no one has tested it yet. Once someone confirms it is working then I'll happily put it in the master branch and make sure it's Python 2&3 compatible.
